packages/arb-bridge-eth/contracts/bridge/NitroMigrator.sol

```diff
@@ -277,9 +274,10 @@ contract NitroMigrator is Ownable, IMessageProvider {
         }
         bridge.setOutbox(address(this), false);
 
-        // the bridge will proxy executeCall calls from the classic outboxes
-        nitroBridge.setOutbox(address(bridge), true);
-        bridge.setReplacementBridge(address(nitroBridge));
+        address[] memory oldOutboxes = bridge.setReplacementBridge(IBridge(address(nitroBridge)));
+        for (uint256 i = 0; i < oldOutboxes.length; i++) {
+            nitroBridge.setOutbox(oldOutboxes[i], true);
+        }
```

Comment:
if the migrator is the owner of the old and new bridge, I think we should have the logic from `setReplacementBridge` done here (so all the changes happen on the same place)
```suggestion
        address[] memory oldOutboxes = bridge.allowedOutboxList();
        for (uint256 i = 0; i < oldOutboxes.length; i++) {
            IOutbox(outbox).setBridge(address(nitroBridge));
            bridge.setOutboxImpl(outbox, false);
            nitroBridge.setOutbox(oldOutboxes[i], true);
        }
```